patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -141,6 +209,19 @@
 +   private net.minecraftforge.common.ToolType harvestTool;
 +   private int harvestLevel;
 +   private final net.minecraftforge.common.util.ReverseTagWrapper<Block> reverseTags = new net.minecraftforge.common.util.ReverseTagWrapper<>(this, BlockTags::getGeneration, BlockTags::func_199896_a);
++   // For internal use only, captures block drops before they're added to the world so they can be tweaked with events.
++   protected static ThreadLocal<Boolean> captureDrops = ThreadLocal.withInitial(() -> false);
```

Comment:
Thread locals are extremely performance intensive. So limiting them as much as possible is how to go.
And that is why I implemented the old version like I did, with a single threadlocal and using != null as the capture check.
as well as a single `List<ItemStack> capture(List<ItemStack> lst)` 

It is also probably worth patching the spawnAsEntity calls in the subblocks to call more contextual versions of the spawn functions. As that'd be a smaller patch, as well as bypassing the need for thread locals.

Reply:
Fair point on the ThreadLocals, I just did things this way to make things a bit more readable. Performance is more important though, so I'll make some changes there.

Not sure what you mean about patching the spawnAsEntity method calls in subblocks though, do you mean just adding a method to block that captures the drops and finishes capturing the drops instead of calling the start and stop capture methods for each patch?

Reply:
My point is why even attempt to spawn the item in the world/capture it in a thread local. Just... make a list with the ItemStack and fire the events. Bypass all of the thread local things and spawnAsEntity call directly.